File: library/src/main/java/cn/wandersnail/widget/viewpager/BaseFragmentPagerAdapter.java
package cn.wandersnail.widget.viewpager;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.PagerAdapter;

import java.util.List;

/**
 * 基本ViewPager的基类
 * <p>
 * date: 2019/8/22 23:26
 * author: zengfansheng
 */
public class BaseFragmentPagerAdapter extends FragmentPagerAdapter {
    private List<Fragment> fragments;

    public BaseFragmentPagerAdapter(FragmentManager fm, @NonNull List<Fragment> fragments) {
        super(fm);
        this.fragments = fragments;
    }

    @Override
    public Fragment getItem(int position) {
        return fragments.get(position);
    }

    @Override
    public int getCount() {
        return fragments.size();
    }

    @Override
    public int getItemPosition(@NonNull Object object) {
        return PagerAdapter.POSITION_NONE;
    }
}

File: library/src/main/java/com/snail/widget/viewpager/BaseFragmentPagerAdapter.kt
package com.snail.widget.viewpager

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.PagerAdapter

/**
 * 描述: 基本ViewPager的基类
 * 时间: 2018/8/26 22:57
 * 作者: zengfansheng
 */
open class BaseFragmentPagerAdapter(fm: FragmentManager, private val fragments: MutableList<out Fragment>) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return fragments[position]
    }

    override fun getCount(): Int {
        return fragments.size
    }

    override fun getItemPosition(any: Any): Int {
        return PagerAdapter.POSITION_NONE
    }
}
